clients/js/src/cipher.ts

```diff
@@ -1,168 +1,200 @@
+  public encryptCall(calldata?: BytesLike | null): BytesLike {
+    // Txs without data are just balance transfers, and all data in those is public.
+    if (calldata === undefined || calldata === null || calldata.length === 0)
+      return '';
 
-  /** Encrypts the plaintext and formats it into an envelope. */
-  public async encryptEnvelope(
-    plaintext?: BytesLike,
-  ): Promise<Envelope | undefined> {
-    if (plaintext === undefined) return;
-    if (!isBytesLike(plaintext)) {
+    if (!isBytesLike(calldata)) {
       throw new Error('Attempted to sign tx having non-byteslike data.');
     }
-    if (plaintext.length === 0) return; // Txs without data are just balance transfers, and all data in those is public.
-    const { data, nonce } = await this.encryptCallData(getBytes(plaintext));
-    const pk = this.publicKey;
-    const epoch = this.epoch;
-    const body = pk.length && nonce.length ? { pk, nonce, data, epoch } : data;
-    if (this.kind === Kind.Plain) return { body };
-    return { format: this.kind, body };
-  }
 
-  protected async encryptCallData(
-    plaintext: Uint8Array,
-  ): Promise<AeadEnvelope> {
-    const body = cbor.encode({ body: plaintext });
-    const { ciphertext: data, nonce } = await this.encrypt(body);
-    return { data, nonce };
-  }
+    const innerEnvelope = cborEncode({ body: getBytes(calldata) });
 
-  /**
-   *  Decrypts the data contained within call
-   *
-   *  This is useful for creating tools, and also decoding
-   *  previously-sent transactions that have used the same
-   *  encryption key.
-   */
+    const { ciphertext, nonce } = this.encrypt(innerEnvelope);
 
-  public async decryptCallData(
-    nonce: Uint8Array,
-    ciphertext: Uint8Array,
-  ): Promise<Uint8Array> {
-    return cbor.decode(await this.decrypt(nonce, ciphertext)).body;
-  }
+    const envelope: Envelope = {
+      format: this.kind,
+      body: {
+        pk: this.publicKey,
+        nonce,
+        data: ciphertext,
+        epoch: this.epoch,
+      },
+    };
 
-  /**
-   * @hidden Encrypts a CallResult in the same way as would be returned by the runtime.
-   * This method is not part of the SemVer interface and may be subject to change.
-   */
-  public async encryptCallResult(
-    result: CallResult,
-    reportUnknown = false,
-  ): Promise<Uint8Array> {
-    if (result.fail) return cbor.encode(result);
-    const encodedResult = cbor.encode(result);
-    const { ciphertext, nonce } = await this.encrypt(encodedResult);
-    const prop = reportUnknown ? 'unknown' : 'ok';
-    return cbor.encode({ [prop]: { nonce, data: ciphertext } });
+    return asBytesLike(calldata, cborEncode(envelope));
   }
 
-  /** Decrypts the data contained within a hex-encoded serialized envelope. */
-  public async decryptEncoded(callResult: BytesLike): Promise<string> {
-    return hexlify(
-      await this.decryptCallResult(cbor.decode(getBytes(callResult))),
-    );
+  public decryptCall(envelopeBytes: BytesLike): BytesLike {
```

Comment:
Test only function or do we expect developers to use this while debugging etc?

Reply:
At the moment it's only used for tests, to ensure roundtrip of encryption/decryption, we also don't use the `encryptResult` outside of tests.

It's not very useful for end-user devs while debugging, as the ephemeral keypair used for encrypting the transactions gets discarded or is otherwise inaccessible.

I guess we could move this into a test utils file?